xray_test.go

```diff
@@ -0,0 +1,196 @@
+	"github.com/jfrog/jfrog-cli-core/v2/utils/coreutils"
+	xrutils "github.com/jfrog/jfrog-cli-core/v2/xray/utils"
+	"github.com/jfrog/jfrog-cli/utils/tests"
+	"github.com/jfrog/jfrog-client-go/auth"
+	clientutils "github.com/jfrog/jfrog-client-go/utils"
+	"github.com/jfrog/jfrog-client-go/utils/io/fileutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/httputils"
+	"github.com/jfrog/jfrog-client-go/utils/version"
+	"github.com/jfrog/jfrog-client-go/xray/services"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	xrayEndpoint = "xray/"
+)
+
+var (
+	xrayDetails     *config.ServerDetails
+	xrayAuth        auth.ServiceDetails
+	xrayHttpDetails httputils.HttpClientDetails
+	// JFrog CLI for Xray commands
+	xrayCli *tests.JfrogCli
+)
+
+func InitXrayTests() {
+	initXrayCli()
+}
+
+func authenticateXray() string {
+	xrayDetails = &config.ServerDetails{XrayUrl: clientutils.AddTrailingSlashIfNeeded(*tests.JfrogUrl) + xrayEndpoint}
+	cred := "--url=" + xrayDetails.XrayUrl
+	if *tests.JfrogAccessToken != "" {
+		xrayDetails.AccessToken = *tests.JfrogAccessToken
+		cred += fmt.Sprintf(" --access-token=%q", xrayDetails.AccessToken)
+	} else {
+		xrayDetails.User = *tests.JfrogUser
+		xrayDetails.Password = *tests.JfrogPassword
+		cred += fmt.Sprintf(" --user=%q --password=%q", xrayDetails.User, xrayDetails.Password)
+	}
+
+	var err error
+	if xrayAuth, err = xrayDetails.CreateXrayAuthConfig(); err != nil {
+		coreutils.ExitOnErr(errors.New("Failed while attempting to authenticate with Xray: " + err.Error()))
+	}
+	xrayDetails.XrayUrl = xrayAuth.GetUrl()
+	xrayHttpDetails = xrayAuth.CreateHttpClientDetails()
+	return cred
+}
+
+func initXrayCli() {
+	if xrayCli != nil {
+		return
+	}
+	cred := authenticateXray()
+	xrayCli = tests.NewJfrogCli(execMain, "jfrog xr", cred)
+}
+
+// Tests basic binary scan by providing pattern (path to testdata binaries) and --licenses flag
+// and asserts any error.
+func TestXrayBinaryScan(t *testing.T) {
+	initXrayTest(t, xrutils.GraphScanMinVersion)
+	newStdout, stdWriter, previousStdout := tests.RedirectStdOutToPipe()
+	// Restore previous stdout when the function returns
+	defer func() {
+		os.Stdout = previousStdout
+	}()
+
+	binariesPath := filepath.Join(filepath.FromSlash(tests.GetTestResourcesPath()), "xray", "binaries", "*")
+	runXr(t, "scan", "--licenses", "--format=json", binariesPath)
+	// Closing the temp stdout in order to be able to read it's content.
+	stdWriter.Close()
+	verifyScanResults(t, newStdout, 0, 1, 1, previousStdout)
+}
+
+// Tests npm audit by providing simple npm project and asserts any error.
+func TestXrayAuditNpm(t *testing.T) {
+	initXrayTest(t, xrutils.GraphScanMinVersion)
+	newStdout, stdWriter, previousStdout := tests.RedirectStdOutToPipe()
+	// Restore previous stdout when the function returns
+	defer func() {
+		os.Stdout = previousStdout
+	}()
+	tempDirPath, err := fileutils.CreateTempDir()
+	assert.NoError(t, err)
+	defer fileutils.RemoveTempDir(tempDirPath)
```

Comment:
The above line should be changed to:
```
defer assert.NoError(t, fileutils.RemoveTempDir(tempDirPath))
```
The code includes a few other places where ```RemoveTempDir``` doesn't include error check, so let's fix them as well.
The same goes for the ```defer os.Chdir(prevWd)``` line below in this function.